libraries/core_libs/network/src/tarcap/threadpool/packets_blocking_mask.cpp

```diff
@@ -76,6 +76,28 @@ std::optional<taraxa::level_t> PacketsBlockingMask::getSmallestDagLevelBeingProc
   return {};
 }
 
+void PacketsBlockingMask::setDagBlockBeingProcessed(const PacketData& packet) {
+  // Signature is used as id for the dag block since it is cheaper to get signature than to calculate hash at this point
+  // and signature should be unique per dag block
+  sig_t sig = DagBlock::extract_signature_from_rlp(packet.rlp_);
+
+  // If blocking is working correctly it should not be possible that we already are processing the same block
+  const auto processing_dag_block = processing_dag_blocks_.find(sig);
+  assert(processing_dag_block == processing_dag_blocks_.end());
```

Comment:
also I would move find inside assert as for Release it will need to result of find

Reply:
Fixed